### ISSUE TITLE
Add linear gradient descent

### ISSUE THREAD

**Issue by binuud** [binuud/ai-green-field#11]
- add a plotter class to draw graph
- add a basic neural network class
- add a class for linear regression, with loss calculation
